source/extensions/filters/common/set_filter_state/filter_config.cc

```diff
@@ -1,13 +1,27 @@
 #include "source/extensions/filters/common/set_filter_state/filter_config.h"
 
+#include "envoy/registry/registry.h"
+
 #include "source/common/formatter/substitution_format_string.h"
+#include "source/common/router/string_accessor_impl.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Filters {
 namespace Common {
 namespace SetFilterState {
 
+class GenericStringObjectFactory : public StreamInfo::FilterState::ObjectFactory {
+public:
+  std::string name() const override { return "envoy.string"; }
```

Comment:
nit: prefer an `envoy.filter_states.string` to show the category of the factory.

Reply:
The factory category is `filter_state.object` so it would be awkward either way I think. Maybe we can rationalize it later? I like `envoy.string`  because it's short and can be used in other places (matcher data, for example).